python/src/scipp/coords.py

```diff
@@ -0,0 +1,269 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+# @author Simon Heybrock, Jan-Lukas Wynen
+
+import inspect
+from typing import Union, List, Dict, Tuple, Callable
+from . import Variable, DataArray, Dataset, bins, VariableError
+
+
+def _argnames(func):
+    spec = inspect.getfullargspec(func)
+    if spec.varargs is not None or spec.varkw is not None:
+        raise ValueError(
+            "Function with variable arguments not allowed in conversion graph."
+        )
+    return spec.args + spec.kwonlyargs
+
+
+class Graph:
+    def __init__(self, graph):
+        # Keys in graph may be tuple to define multiple outputs
+        self._graph = {}
+        for key in graph:
+            for k in [key] if isinstance(key, str) else key:
+                self._graph[k] = graph[key]
+
+    def __getitem__(self, name):
+        return self._graph[name]
+
+    def __contains__(self, name):
+        return name in self._graph
+
+    def show(self, size=None):
+        from graphviz import Digraph
```

Comment:
It would be nice to catch this if graphviz is not installed and provide a better error message telling the user why the package is needed and what to do about it.

Also, I feel like this should be added to the developer environment YAML file.

Reply:
Done already!